### ISSUE TITLE
Add support for scoring system

### ISSUE THREAD

**Issue by TommySoucy** [TommySoucy/H3MP#197]

**Comment by TommySoucy** [comment]
We could maybe even add an MP leaderboard if all we gotta do is call the steamworks API with a custom game ID

**Comment by redwingsx62-sys** [comment]
Hi, I ran into this while trying to run solo TNH with a player body mod installed (H3VRPlayerBodySystem → depends on H3MP as a hard dependency). Wanted to share what I found in case it's useful for the scoring rework mentioned in #251.

Root cause: GeneralPatches.cs → ScorePatch.Prefix() unconditionally returns false, which skips TNH_ScoreDisplay.SubmitScoreAndGoToBoard() 
entirely — regardless of whether Mod.managerObject is actually set (i.e. regardless of whether a multiplayer session is active).

This means any player who installs a body mod (or any other mod that pulls in H3MP as a dependency) loses the ability to submit solo TNH 
leaderboard scores, even with zero multiplayer activity going on.

I wrote a small standalone BepInEx plugin that waits for H3MP's patches to be applied, then unpatches ScorePatch.Prefix and replaces 
it with a version that checks Mod.managerObject before deciding whether to block submission:

- managerObject == null (solo) → let vanilla SubmitScoreAndGoToBoard run
- managerObject != null (in a session) → keep blocking, same as current behavior

It's working well for me in testing. Before I publish it on Thunderstore, I wanted to check:

1. Is there a reason the block is unconditional rather than session-gated? (Anti-cheat concern about score integrity even outside active sessions, maybe?)
2. Would you be open to this kind of condition check being merged into H3MP directly, possibly as part of the scoring rework in #251? Happy to share the patch/PR if useful.

Thanks for the mod either way, it's been great for solo play with body mods.
